src/functions-and-arrays.js

```diff
@@ -1,41 +1,122 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(num1,num2) {
+  if (num1>num2){
+  return num1
+ } else if (num2>num1){
+  return num2
+ } else {
+  return num1 || num2 
+ }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
 
+function findLongestWord(exerciseArray){
+  let longestWord = "";
+  if (exerciseArray.length === 0){
+    return null;
+  } else if(exerciseArray.length === 1){
+    return exerciseArray[0]
+  } else {
+    for (let i = 0; i < exerciseArray.length; i++) {
+      if(longestWord.length < exerciseArray[i].length) {
+        longestWord = exerciseArray[i];
+      }
+    }
+    return longestWord;
+  }
+} 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+let total = 0
+function sumNumbers(exerciseNumber) {
+  
+  if (exerciseNumber.length === 0){
+    return 0;
+  } let sum = 0;
+  for ( let i = 0; i < exerciseNumber.length; i +=1){
+    sum += exerciseNumber[i]
+  }
+   return sum
+
+}
+  
+
+    
+
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(bonus) {
+  if (bonus.length === 0){
+    return 0;
+  } let sum = 0;
+  for (let i = 0; i < bonus.length; i++){
+    sum += bonus[i]
+  }
+  return sum
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(arr) {
+  if (arr.length === 0){
+    return null
+  } let avrArr = 0
```

Comment:
Since you previously created a function to sum numbers, you should reuse it here instead of repeating the whole logic to sum the numbers.
```
let avrArr = sumNumbers(arr);
return avrArr / arr.length
```